students/JeroenSweerts/Lesson9/mailroom5.py

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env python3
+
+class Donor():
+    def __init__(self, name=None):
+        self._name = name
+        self._donations = []
+
+    @property
+    def name(self):
+        """Property for the donor's name."""
+        return self._name
+
+    @name.setter
+    def name(self, val):
+        """Setter for the donor's name."""
+        self._name = val
+
+    @property
+    def donations(self):
+        """Property for the donor's list of donations."""
+        return self._donations
+
+    @donations.setter
+    def donations(self, val):
+        """Setter for the donor's list of donations."""
+        self._donations = val
+
+
+class Donor_Collection(dict):
+    def __init__(self, donors):
+        self._donors = donors
+
+    @property
+    def donors(self):
+        """Property for the Donor_Collection's list of donors."""
+        return self._donors
+
+    @donors.setter
+    def donors(self, val):
+        """Setter for the Donor_Collection's list of donors."""
+        self._donors = val
+
+    def add_donor(self, donor):
+        """Add a new donor to the collection."""
+        self.donors[donor.name] = donor.donations
+
+    def donate(self, name, donation):
+        """Add a donation to the list of donations for a specific donor.
+        If the donor doesn't exist yet, then add it to the donor collection first."""
+        if name not in self.donors.keys():
+            new_donor = Donor(name)
```

Comment:
This seems to be the only place that you're using the `Donor` class, and all it's doing is passing the donor's name into the `donor` dict, so as this is written, you really don't need the `Donor` class at all. You should think about reworking you code to encapsulate all the donor-related logic inside of the `Donor` class. This might include storing and accessing donations, as well as calculating the donation values that you reference in the report.